modules/indexer/issues/dboptions.go

```diff
@@ -17,6 +17,10 @@ func ToSearchOptions(keyword string, opts *issues_model.IssuesOptions) *SearchOp
 		IsClosed:  opts.IsClosed,
 	}
 
+	if opts.ProjectID != 0 {
+		searchOpt.ProjectID = &opts.ProjectID
+	}
+
```

Comment:
if opts.ProjectID != 0
searchOpt.ProjectID = &opts.ProjectID
if opts.ProjectID == 0
searchOpt.ProjectID = &opts.ProjectID = 0

So why not add this into searchOpt directly?

Reply:
I didn't get past the CI by doing it this way at first, so I'll change it to this and try again.

Reply:
add this into searchOpt directly can not  past the CI

Reply:
So why CI failed. I will also check this.

Reply:
maybe about `nil` and `0`

Reply:
ProjectID is a pointer, the default value is `nil` 😕

Reply:
right, `0` is not `nil `, so we can't add it  into `searchOpt` directly

Reply:
Different to other places, comes from #26012
Not bad, but need to pay attention. 😄